src/App.tsx

```diff
@@ -1,156 +1,33 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useContext } from 'react';
+import { TodosContext } from './context/ToDosContext';
+import { SortType } from './types/types';
+import { Footer } from './components/Footer/Footer';
+import { TodoList } from './components/ToDoList/ToDoList';
+import { Header } from './components/Header/Header';
 
 export const App: React.FC = () => {
+  const { todos, sorted } = useContext(TodosContext);
+
+  const visibleTodos = (() => {
+    switch (sorted) {
+      case SortType.Active:
+        return todos.filter(todo => !todo.completed);
+      case SortType.Completed:
+        return todos.filter(todo => todo.completed);
+      default:
+        return todos;
+    }
+  })();
```

Comment:
Wrap `visibleTodos` in a `useMemo` to avoid unnecessary reprocessing

Reply:
Done!